src/main/java/com/redhat/maven/index/checker/MavenIndexChecker.java

```diff
@@ -267,8 +272,26 @@ public void transferCompleted(TransferEvent transferEvent) {
 
         logger.info("Reading index");
         JSONArray results = new JSONArray();
-        Set<OutputInfo> info = new HashSet<OutputInfo>();
+        File dbFile = File.createTempFile("mic", "db");
+        dbFile.deleteOnExit();
+        /* We use a sqlite on-disk db to be able to leave out duplicates
+           while not keeping all the package objects in-memory.
+           The duplicates exist since the lucene index contains
+           all artifacts for every package version, e.g. not just the
+           binary jar, but also javadocs, sources, ... (if present). */
```

Comment:
When I was playing with this I saw that `ArtifactInfo.getClassifier()` returns one of `{null | javadoc | sources}` so I'm curious whether simple `"jar".equals(ai.getPackaging()) && "sources".equals(ai.getClassifier())` wouldn't also remove the duplicates ?